### PR TITLE
feat: Groq Inference Extension

### PR DESCRIPTION
**Pull request by metaspartan**
## Describe Your Changes

- Adds a new extension and functionality for inference with the new Groq APIs

This is the implementation of https://github.com/janhq/jan/issues/2229

## Self Checklist

- [✔️ ] Added groq inference extension
- [x] Updated docs (for new feature)
- [x] Added new default mixtral-8x7b and llama2-70b models from Groq

### CODE REVIEW

**Comment by metaspartan** [comment]
Will this ever be reviewed/ack? It has been up for a week now...

File: core/src/types/model/modelEntity.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export type ModelInfo = {
 export enum InferenceEngine {
   nitro = 'nitro',
   openai = 'openai',
+  groq = 'groq',
```
Comment (louis-jan) [inline core/src/types/model/modelEntity.ts:21]: Thank you for the quick hack, we will refactor these engines soon.

**Review by louis-jan — APPROVED** [review]
LGTM

**Review by louis-jan — APPROVED** [review]

**Comment by louis-jan** [comment]
Just a few linter issues left. Please help fix so we can proceed. @metaspartan

**Comment by metaspartan** [comment]
> Just a few linter issues left. Please help fix so we can proceed. @metaspartan

Resolved ✔️

**Comment by Van-QA** [comment]
Linking to #2227

**Pull request merged**